src/component.cpp

```diff
@@ -665,7 +665,7 @@ static void displayCompObj(DROID *psDroid, bool bButton, const glm::mat4 &viewMa
 					localModelMatrix *= glm::rotate(UNDEG(65536 / 2), glm::vec3(0.f, 0.f, 1.f));
 				}
 
-				localModelMatrix *= glm::translate(psShapeBody->connectors[0].xzy);
+				localModelMatrix *= glm::translate(glm::vec3{ psShapeBody->connectors[0].x, psShapeBody->connectors[0].z, psShapeBody->connectors[0].y });
```

Comment:
Seems like this could be:
`glm::translate(glm::vec3( psShapeBody->connectors[0].xzy() ))`